src/main/java/lotto/util/ProfitCalculator.java

```diff
@@ -1,18 +1,16 @@
 package lotto.util;
 
+import lotto.constants.Rank;
 import lotto.domain.LottoResult;
 
 import static lotto.domain.PayAmount.LOTTO_PRICE;
 
 public class ProfitCalculator {
 
-    public static final int[] PRIZE_MONEY = {0, 0, 0, 5000, 50000, 1500000, 2000000000};
-    public static final int MIN_WINNING_NUM = 3;
-
     public static double calculateProfit(LottoResult lottoResult, int lottoAmount) {
         double profit = 0;
-        for (int i = MIN_WINNING_NUM; i < PRIZE_MONEY.length; i++) {
-            profit += lottoResult.getLottoResult(i) * PRIZE_MONEY[i];
+        for (Rank rank : lottoResult.getLottoResult().keySet()) {
+            profit += rank.getWinningMoney() * lottoResult.getLottoResult().getOrDefault(rank, 0);
```

Comment:
데이터를 외부에서 가져와서 판단하는것이 아닌 
데이터를 가진 객체에게 메시지를 전달해보면 어떨까요?
```suggestion
         profit += lottoResult.profit(rank);
```
```
#LottoResult
public double profit(Rank rank) {
      return rank.winningMoney(lottoResult.getOrDefault(rank, 0));
}
```

[[객체지향 생활체조 원칙] 규칙 9. getter/setter/property를 쓰지 않는다](https://limdingdong.tistory.com/15)

참고해주세요!

Reply:
사실 이부분은.. 
LottoResult 객체에 책임을 부여하여 해당 클래스 내에서 수익률을 계산 및 반환할 수 있도록 처음에 구현을 했었는데요. Step3 리뷰에서 리뷰어님이 별도의 객체로 분리하는게 나을 것 같다는 의견을 주셔서 분리했던 내용입니다. 당시에도 사실 별도 객체로 분리하기 보단 객체에 책임을 주는게 나은 방향이 아닌가 싶긴 했는데 일단 리뷰어님 말씀에 따라 진행했습니다 😢  
최초 PR 에서 구조화가 너무 엉성하여 리뷰어님께서 리뷰를 하시다 잘못 말씀해주신걸수도 있고, 제가 잘못이해한걸수도 있겠네요..
다시 객체에 책임을 부여하도록 하겠습니다! 😊

Reply:
최대한 객체에 책임을 부여한다고 고민하며 구현한건데,
해당 부분도 객체 내에서 처리할 수 있었겠네요. 
생각이 짧았습니다.
해당부분 수정했습니다. 
감동의 소스리뷰입니다. 감사합니다 🙏